### PR TITLE
Allow the module to properly import on case-sensitive OSes

### PR DESCRIPTION
**Pull request by DWSR**

### CODE REVIEW

File: PowerShellHumanizer.psm1 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Add-Type -Path "$PSScriptRoot\humanizer.dll"
```
Comment (vexx32) [inline PowerShellHumanizer.psm1]: This is still fragile and doesn't account for the .DLL file to be potentially named in any mixed case or upper case as may happen in some circumstances.

It would be more pragmatic to do something like this:

```powershell
$ModuleDll = Get-ChildItem -Path $PSScriptRoot |
    Where-Object Name -eq 'humanizer.dll' |
    ForEach-Object FullName

Add-Type -Path $ModuleDll
```

Comment (DWSR) [reply PowerShellHumanizer.psm1]: I'm not sure what circumstances you're referring to given that the DLL is directly committed to the repository. Could you elaborate?

Comment (vexx32) [reply PowerShellHumanizer.psm1]: The likelihood isn't incredibly high, but it's not impossible that the DLL be rebuilt and updated in the future under a different naming or the same name with different casing, or just that it gets renamed by accident in future.

To avoid unnecessary later commits to amend this same problem, it seems wisest to simply head it off now and avoid hard-to-anticipate dependencies by explicitly evading the issue in the code.

Comment (SteveL-MSFT) [reply PowerShellHumanizer.psm1]: For PowerShell projects, we don't spend effort covering this corner case.  Instead, we spend effort running tests on both Windows and non-Windows platforms so in the unlikely case an assembly's casing gets changed, we will catch it in our tests and handle it appropriately.

Comment (dfinke) [reply PowerShellHumanizer.psm1]: I'll setup __tests__ and wire in Azure DevOps. @DWSR and @vexx32, will leave it there for PRs.
